superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx

```diff
@@ -299,19 +317,42 @@ const displayField = ({
   getValidation,
   validationErrors,
   db,
+  setPublicSheets,
 }: FieldPropTypes) => (
-  <ValidatedInput
-    id="database_name"
-    name="database_name"
-    required
-    value={db?.database_name}
-    validationMethods={{ onBlur: getValidation }}
-    errorMessage={validationErrors?.database_name}
-    placeholder=""
-    label="Display Name"
-    onChange={changeMethods.onChange}
-    helpText={t('Pick a nickname for this database to display as in Superset.')}
-  />
+  <>
+    <ValidatedInput
+      id="database_name"
+      name="database_name"
+      required
+      value={db?.database_name}
+      validationMethods={{ onBlur: getValidation }}
+      errorMessage={validationErrors?.database_name}
+      placeholder=""
+      label="Display Name"
+      onChange={changeMethods.onChange}
+      helpText={t(
+        'Pick a nickname for this database to display as in Superset.',
+      )}
+    />
+
+    {db?.engine === 'gsheets' && (
+      <>
+        <FormLabel required>{t('Type of Google Sheets Allowed')}</FormLabel>
+        <Select
+          style={{ width: '100%' }}
+          onChange={(value: string) => setPublicSheets(value)}
```

Comment:
if you're not manipulating the arguments at all, you should be able to just pass the function itself, rather than create a new arrow function.

Reply:
When I do this `onChange = {setPublicSheets(value)`} I get this reference error. 

ReferenceError: value is not defined

Reply:
When you write:

```js
// this is a function
(value: string) => setPublicSheets(value)
```

You're defining an anonymous function that takes a string called `value` and calls `setPublicSheets` on it, and you're passing that function to `onChange`.

If you wanted to name that function, you could do one of these:

```js
const func = (value: string) => setPublicSheets(value);

function func(value: string) {
    return setPublicSheets(value);
}
```

But why do you need to create that anonymous function and pass to `onChange`. You **already have** a function that takes a string called `value` and returns the result of passing it to `setPublicSheets`... it's `setPublicSheets` itself!

So all you need to do is:

```js
onChange={setPublicSheets}
```